server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -180,18 +180,13 @@ public List<Pair<DataSegment, String>> retrieveUsedSegmentsAndCreatedDates(Strin
         "SELECT created_date, payload FROM %1$s WHERE dataSource = :dataSource AND used = true"
     );
 
-    boolean hasEternityInterval = false;
-    for (Interval interval : intervals) {
-      if (Intervals.isEternity(interval)) {
-        hasEternityInterval = true;
-        break;
-      }
-    }
+    final boolean compareAsString = intervals.stream().allMatch(Intervals::canCompareEndpointsAsStrings);
```

Comment:
Please rename to `compareIntervalsAsString` or `compareIntervalEndpointsAsString`.

Reply:
Done